dcos/client.go

```diff
@@ -112,7 +112,7 @@ func (a *APIClient) CurrentDCOSConfig() Config {
 
 // HTTPClient returns a http.Client which does authenticated requests to DC/OS
 func (a *APIClient) HTTPClient() *http.Client {
-	return NewHTTPClient(a.dcosConfig)
+	return a.cfg.HTTPClient
```

Comment:
do we really want to use the same http.Client? as its a pointer one can control and change the http.Client used by the module. 

I especially tried to avoid this. As changes to this client would lead to unexpected behavior of the dcos module.

Reply:
The idea is to just turn a new http.Client build from the modules used config